src/Utilities/Idm/mf6blockfile/StructArray.f90

```diff
@@ -280,15 +287,46 @@ end subroutine allocate_charstr_type
   !> @brief allocate int1d input type
   !<
   subroutine allocate_int1d_type(this, sv)
+    use ConstantsModule, only: LENMODELNAME
+    use MemoryHelperModule, only: create_mem_path
+    use SimVariablesModule, only: idm_context
     class(StructArrayType) :: this !< StructArrayType
     type(StructVectorType), intent(inout) :: sv
     integer(I4B), dimension(:, :), pointer, contiguous :: int2d
     type(STLVecInt), pointer :: intvector
-    integer(I4B), pointer :: ncelldim
+    integer(I4B), pointer :: ncelldim, exgid
+    character(len=LENMEMPATH) :: input_mempath
+    character(len=LENMODELNAME) :: mname
+    type(CharacterStringType), dimension(:), contiguous, &
+      pointer :: charstr1d
     !
     if (sv%idt%shape == 'NCELLDIM') then
       !
-      call mem_setptr(ncelldim, sv%idt%shape, this%component_mempath)
+      ! -- if EXCHANGE set to NCELLDIM of appropriate model
+      if (this%mf6_input%component_type == 'EXG') then
```

Comment:
Agreed, it would be best to keep all of the lower level code free of all package specific handling.  I'll plan to revisit this soon- it is a good reminder as we head towards advanced and subpackage support.